core/domain/topic_services.py

```diff
@@ -91,46 +91,35 @@ def get_all_skill_ids_assigned_to_some_topic():
 
 def get_all_skill_ids_assigned_to_some_topic_with_topic_details():
```

Comment:
Can you get rid of this function and inline it in its caller? We should not be passing dicts around.

The current complexity seems fine since you're just using one get_all() to the datastore, I think. (If this was the case before then that means I got it wrong in the previous review, sorry about that.)

Reply:
Removed. Done.